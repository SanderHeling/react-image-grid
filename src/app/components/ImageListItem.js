import React from 'react';
import PropTypes from 'prop-types';


function ImageListItem({ src, alt, onClick }) {
    return (
        <div className="image-list__item">
            <img src={src} alt={alt} onClick={e => onClick(e)} />
        </div>
    );
}

ImageListItem.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string,
    onClick: PropTypes.func,
};

ImageListItem.defaultProps = {
    alt: '',
    onClick: () => {},
};

export default ImageListItem;
