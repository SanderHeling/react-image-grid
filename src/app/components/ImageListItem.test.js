import React from 'react';
import renderer from 'react-test-renderer';

import ImageListItem from './ImageListItem';


test('should render correctly', () => {
    const component = renderer.create(
        <ImageListItem src="https://source.unsplash.com/random/300x500" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
