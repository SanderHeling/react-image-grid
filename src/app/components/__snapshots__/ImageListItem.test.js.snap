// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="image-list__item"
>
  <img
    alt=""
    onClick={[Function]}
    src="https://source.unsplash.com/random/300x500"
  />
</div>
`;
